frontend/src/components/NotFound.test.tsx

```diff
@@ -0,0 +1,30 @@
+import { describe, expect, test } from "vitest";
+import { render, screen } from "@testing-library/react";
+import { MemoryRouter, Route, Routes } from "react-router";
+import NotFound from "./NotFound";
+
+interface NotFoundTestParams {
+  path: string;
+  expectedText: string;
+}
+
+const testCases: NotFoundTestParams[] = [
+  {
+    path: "/random-path",
+    expectedText: "404 Not Found",
+  },
+];
+
+describe.each(testCases)("NotFound Component", ({ path, expectedText }) => {
```

Comment:
I'm not sure I see a reason for the `describe.each` -- I can't think of a reason we'd need a second test here, there's one single behavior (visiting a non-existing route) that we're testing.
So I'd just inline the "array" and create a single normal test.